### ISSUE TITLE
Use Functional React components

### ISSUE THREAD

**Issue by Leif-Frederiksen** [pnp/generator-spfx#249]
#### General Information
* [ ] Usage
* [ ] Development
* [ ] Documentation
* [ X] Feature Request

#### Specific generator
* [ ] Addons
* [ ] HandlebarsJS
* [X ] React

Since I switched to functional React components and hooks, I really don't ever want to look back to the old-school class based components.

I would like to have the default scaffolded code to be functional instead of class based.

**Comment by StfBauer** [comment]
Hi @Leif-Frederiksen - if you are able to provide a sample solution that has templates for all base assets you can create we can make this happen.

I guess the only thing that really makes sense is web part. What do you think?

We can scaffold those together even if you like.

**Comment by StfBauer** [comment]
@Leif-Frederiksen Can you provide such a sample?

**Comment by Leif-Frederiksen** [comment]
Hi Stefan,

After having tried to find heads and tails in how the generatorsystem
works, I think I'm going to have to fold. I guess the actual content of the
component is not in the PnP template, but inherited for MS's template?
Anyways, the resulting code I thought could be good for motivating for
writing functional components is something like this:
import * as React from 'react';
import styles from './HelloWorld.module.scss';
import { IHelloWorldProps } from './IHelloWorldProps';
import { escape } from '@microsoft/sp-lodash-subset';

export default function HelloWorld(props:IHelloWorldProps) {
  return (
    <div className={ styles.helloWorld }>
      <div className={ styles.container }>
        <div className={ styles.row }>
          <div className={ styles.column }>
            <span className={ styles.title }>Welcome to SharePoint!</span>
            <p className={ styles.subTitle }>
Customize SharePoint experiences using Web Parts.</p>
            <p className={ styles.description }>{escape(props.description)}
</p>
            <a href="https://aka.ms/spfx" className={ styles.button }>
              <span className={ styles.label }>Learn more</span>
            </a>
          </div>
        </div>
      </div>
    </div>);
}

Regards
Leif

Den ons. 25. mar. 2020 kl. 18.23 skrev Stefan Bauer <
notifications@github.com>:

**Comment by StfBauer** [comment]
@Leif-Frederiksen So it is only the React component that actually needs to get adopted like this?
The only chance is to overwrite after provisioning.

**Comment by Leif-Frederiksen** [comment]
Yes exactly. OK, I was beginning to suspect that. Thanks anyway :-)

Regards
Leif

Den tir. 31. mar. 2020 kl. 12.35 skrev Stefan Bauer <
notifications@github.com>:
